testing/asserts.ts

```diff
@@ -766,8 +771,11 @@ export async function assertRejects<E extends Error = Error>(
     }
     doesThrow = true;
   }
-  if (!doesThrow) {
-    msg = `Expected function to throw${msg ? `: ${msg}` : "."}`;
+  msg = msg ? `: ${msg}` : ".";
+  if (!isPromiseReturned) {
+    throw new AssertionError(`Function throws when expected to reject${msg}`);
+  } else if (!doesThrow) {
```

Comment:
I think it would be better to have this where it was. If an error is thrown instead of rejected, we would incorrectly say "A non-Error object was rejected." instead of "Function throws when expected to reject" like we would before your most recent change.

Reply:
I'm fine with the change to add `: ${msg}` to the end of the message though when msg is set.

Reply:
I'll add a test case that reproduces this unwanted behavior and change the code accordingly then !